router/core/src/main/scala/com/twitter/finagle/buoyant/RetryFilter.scala

```diff
@@ -0,0 +1,85 @@
+package com.twitter.finagle.buoyant
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.service.{RetryBudget, RetryPolicy}
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.finagle.tracing.Trace
+import com.twitter.finagle.{Filter, Service}
+import com.twitter.util._
+
+/**
+ * Cribbed from:
+ * https://github.com/twitter/finagle/blob/develop/finagle-core/src/main/scala/com/twitter/finagle/service/RetryFilter.scala
+ *
+ * Copied here and modified in order to change the type of stats exported by the filter.
+ */
+class RetryFilter[Req, Rep](
+  retryPolicy: RetryPolicy[(Req, Try[Rep])],
+  timer: Timer,
+  statsReceiver: StatsReceiver,
+  retryBudget: RetryBudget
+)
+  extends Filter[Req, Rep, Req, Rep] {
+
+  def this(
+    retryPolicy: RetryPolicy[(Req, Try[Rep])],
+    timer: Timer,
+    statsReceiver: StatsReceiver
+  ) = this(
+    retryPolicy,
+    timer,
+    statsReceiver,
+    RetryBudget()
+  )
+
+  private[this] val retriesStat = statsReceiver.stat("retries")
+
+  private[this] val totalRetries = statsReceiver.scope("retries").counter("total")
+
+  private[this] val budgetExhausted =
+    statsReceiver.scope("retries").counter("budget_exhausted")
```

Comment:
It feels odd to have these scoped like this so that we have
```
foo/retries.count
foo/retries/total
```

Perhaps scope the stat as `statsReceiver.scope("retries").stat("per_request")` so that they end up scoped similarly (also this makes it clearer what the stat is)